dashboard/test/models/user_test.rb

```diff
@@ -1616,6 +1616,44 @@ def track_progress(user_id, script_level, result, pairings: nil)
     assert_equal [], teacher.reload.permissions
   end
 
+  test 'grant admin permission logs to infrasecurity' do
+    teacher = create :teacher
+
+    teacher.stubs(:should_log?).returns(true)
+    ChatClient.
+      expects(:message).
+      with('infra-security',
+        "Granting UserPermission: environment: #{rack_env}, "\
+        "user ID: #{teacher.id}, "\
+        "email: #{teacher.email}, "\
+        "permission: ADMIN",
+        color: 'yellow'
+      ).
+      returns(true)
+
+    teacher.admin = true
+    teacher.save
+  end
+
+  test 'revoke admin permission logs to infrasecurity' do
+    admin_user = create :admin
+
+    admin_user.stubs(:should_log?).returns(true)
+    ChatClient.
+      expects(:message).
+      with('infra-security',
+        "Revoking UserPermission: environment: #{rack_env}, "\
+        "user ID: #{admin_user.id}, "\
+        "email: #{admin_user.email}, "\
+        "permission: ADMIN",
+        color: 'yellow'
+      ).
+      returns(true)
+
+    admin_user.admin = nil
+    admin_user.save
+  end
```

Comment:
Might be worth a test with `should_log?` stubbed to return false and verify `ChatClient.message` is not called.